src/models/sources.ts

```diff
@@ -14,4 +15,20 @@ type SourcesState = {
   currentSource?: SourceWithRepository;
 };
 
+export type CreateSourcePayload = {
```

Comment:
Is it correct? Looks like we use a completely different payload:
```ts
    const body = {
      name,
      repository: {
        type: 'git' as RepositoryTypeEnum,
        uri,
        ...(username ? {usernameSecret: {name: username}} : {}),
        ...(token ? {tokenSecret: {name: token}} : {}),
      },
      namespace,
    };

    createSource(body).then((res: any) => {
      displayDefaultNotificationFlow(res, () => {
        navigate(`/sources/${res.data.metadata.name}`);
      });
    });
```

Reply:
<img width="564" alt="image" src="https://user-images.githubusercontent.com/25991946/234853948-c4197520-b128-48aa-877a-0810dff3843a.png">